Lib/importlib/_bootstrap_external.py

```diff
@@ -312,7 +321,33 @@ def cache_from_source(path, debug_override=None, *, optimization=None):
         if not optimization.isalnum():
             raise ValueError('{!r} is not alphanumeric'.format(optimization))
         almost_filename = '{}.{}{}'.format(almost_filename, _OPT, optimization)
-    return _path_join(head, _PYCACHE, almost_filename + BYTECODE_SUFFIXES[0])
+    filename = almost_filename + BYTECODE_SUFFIXES[0]
+    if sys.bytecode_path is not None:
```

Comment:
Yeah, that seems low-priority to me, given that you could just as easily pass `-X pycache_prefix=foo` when you run `compileall` rather than a new compileall-specific parameter. Maybe there are reasons to have it, but I'd rather leave it as a separate follow-up enhancement.